qt/scientific_interfaces/Indirect/IndirectInterface.cpp

```diff
@@ -50,9 +50,13 @@ IndirectInterface::getInterfaceSettings() const {
 }
 
 void IndirectInterface::manageUserDirectories() {
-  ManageUserDirectories *ad = new ManageUserDirectories(this);
-  ad->show();
-  ad->setFocus();
+  if (!m_manageUserDirectories) {
+    m_manageUserDirectories = std::make_unique<ManageUserDirectories>(this);
+    m_manageUserDirectories->setAttribute(Qt::WA_DeleteOnClose, false);
+  }
+  m_manageUserDirectories->show();
+  m_manageUserDirectories->setFocus();
+  m_manageUserDirectories->setModal(true);
```

Comment:
This doesn't actually make the window modal the first time it's opened, only after closing it and re-opening will it be modal. This can be fixed by calling `setModal` *before* calling `show`. This also means there's no need to call `setFocus` since a modal window will automatically take focus from its parents.